src/SprykerEco/Zed/UnzerGui/Communication/Form/DataProvider/UnzerCredentialsFormDataProvider.php

```diff
@@ -58,22 +59,23 @@ public function getData(?int $idUnzerCredentials = null): UnzerCredentialsTransf
     {
         $unzerCredentialsTransfer = (new UnzerCredentialsTransfer())->setUnzerKeypair(new UnzerKeypairTransfer());
 
-        if (!$idUnzerCredentials) {
-            return $unzerCredentialsTransfer;
+        if ($idUnzerCredentials) {
+            $unzerCredentialsCriteriaTransfer = (new UnzerCredentialsCriteriaTransfer())
+                ->setUnzerCredentialsConditions(
+                    (new UnzerCredentialsConditionsTransfer())->addId($idUnzerCredentials),
+                );
+            $unzerCredentialsCollectionTransfer = $this->unzerFacade->getUnzerCredentialsCollection($unzerCredentialsCriteriaTransfer);
+            $unzerCredentialsTransfer = $unzerCredentialsCollectionTransfer->getUnzerCredentials()[0] ?? $unzerCredentialsTransfer;
```

Comment:
probably it's better to use `->getIteratior()->getCurrent()` instead of pointing exact key